js/src/vector.ts

```diff
@@ -414,6 +414,16 @@ export class DictionaryVector<T extends DataType = DataType> extends Vector<Dict
     }
     public getKey(index: number) { return this.indicies.get(index); }
     public getValue(key: number) { return this.dictionary.get(key); }
+    public reverseLookup(value: T): number|undefined {
+        let key = -1;
+        for (let len = this.dictionary.length; ++key < len;) {
+            if (this.dictionary.get(key) === value) {
```

Comment:
@TheNeuralBit is it worthwhile to cache the dictionary lookup in a local variable to avoid the repeated `this.dictionary` lookups?

Reply:
good call, I'll make that change